snakedwi/workflow/rules/diffusion/reg_dwi_to_t1.smk

```diff
@@ -54,17 +54,94 @@ rule reg_dwi_to_t1:
         ),
     threads: 8
     shell:
-        "greedy -threads {threads} {params.general_opts} {params.rigid_opts} "
-        "-i {input.t1wsynth} {input.avgb0synth} -o {output.xfm_ras} "
-        "&> {log} && "
-        "greedy -threads {threads} {params.general_opts} -rf {input.t1wsynth} "
-        "-rm {input.avgb0} {output.warped_avgb0} -r {output.xfm_ras} &>> {log}"
+        """
+        greedy -threads {threads} {params.general_opts} {params.rigid_opts} \\
+            -i {input.t1wsynth} {input.avgb0synth} -o {output.xfm_ras} \\
+            &> {log}
+
+        greedy -threads {threads} {params.general_opts} -rf {input.t1wsynth} \\
+            -rm {input.avgb0} {output.warped_avgb0} -r {output.xfm_ras} \\
+             &>> {log}
+        """
+
+
+# rule qc_reg_dwi_t1:
+#     input:
+#         ref=bids(
+#             root=work,
+#             suffix="avgb0.nii.gz",
+#             space="T1w",
+#             desc="dwiref",
+#             datatype="dwi",
+#             **subj_wildcards
+#         ),
+#         flo=bids(
+#             root=root,
+#             suffix="T1w.nii.gz",
+#             desc="preproc",
+#             datatype="anat",
+#             **subj_wildcards
+#         ),
+#     output:
+#         png=report(
+#             bids(
+#                 root=root,
+#                 datatype="qc",
+#                 suffix="reg.png",
+#                 **subj_wildcards,
+#                 from_="dwiref",
+#                 to="T1w"
+#             ),
+#             caption="../report/reg_dwi_t1.rst",
+#             category="B0 T1w registration",
+#         ),
+#         html=bids(
+#             root=root,
+#             datatype="qc",
+#             suffix="reg.html",
+#             from_="dwiref",
+#             to="T1w",
+#             **subj_wildcards
+#         ),
+#     group:
+#         "subj"
+#     container:
+#         config["singularity"]["python"]
+#     script:
+#         "../scripts/vis_regqc.py"
+
+rule qc_get_t1_edges:
+    input:
+        brain=rules.n4_t1_withmask.output.t1,
+        mask=bids(
+            root=root,
+            datatype="anat",
+            desc="brain",
+            suffix="mask.nii.gz",
+            **subj_wildcards,
+        )
+    output:
+        temp(
+            bids(
+                work,
+                datatype="anat",
+                desc="cannyEdge",
+                suffix="T1w.nii.gz",
+                **subj_wildcards,
+            )
+        )
+    group: 'subj'
+    container:
+        config["singularity"]["itksnap"]
+    shell:
+        "c3d {input} -times -canny 1mm 10 20 {output}"
 
 
 rule qc_reg_dwi_t1:
     input:
-        ref=rules.reg_dwi_to_t1.output.warped_avgb0,
-        flo=rules.n4_t1_withmask.output.t1,
+        overlay=rules.reg_dwi_to_t1.output.warped_avgb0[0],
```

Comment:
The `[0]` on these lines shouldn't be here, right? Just removing them worked for me in any case..

Reply:
Mmm, I think it depends if the rule we grab it from has a single output or a list of outputs. I end up consuming the item in the script under the assumption it's a str, not a list, and I got burned a couple times during development by failing that assumption. Not sure if the upstream rule has changed though

Reply:
the upstream rule is just a string, so the rule currently seems to grab the first character of the string.. I've reverted this in my recent commit just now but should make sure it still works on your end..

Reply:
Ah okay, so that changed then. Thanks!